tests/chat_completions_detection.rs

```diff
@@ -0,0 +1,950 @@
+            input: vec![],
+            output: vec![],
+        }),
+        warnings: vec![],
+        ..Default::default()
+    };
+
+    // Add detector input mock
+    detector_mocks.mock(|when, then| {
+        when.post()
+            .path(TEXT_CONTENTS_DETECTOR_ENDPOINT)
+            .json(ContentAnalysisRequest {
+                contents: vec!["Hi there!".into()],
+                detector_params: DetectorParams::new(),
+            });
+        then.json(vec![Vec::<ContentAnalysisResponse>::new()]);
+    });
+    // Add detector output mock
+    detector_mocks.mock(|when, then| {
+        when.post()
+            .path(TEXT_CONTENTS_DETECTOR_ENDPOINT)
+            .json(ContentAnalysisRequest {
+                contents: vec!["Hello!".into()],
+                detector_params: DetectorParams::new(),
+            });
+        then.json(vec![Vec::<ContentAnalysisResponse>::new()]);
+    });
+
+    // Add chat completions mock
+    chat_mocks.mock(|when, then| {
+        when.post()
+            .path(CHAT_COMPLETIONS_ENDPOINT)
+            .json(ChatCompletionsRequest {
+                messages: messages.clone(),
+                model: MODEL_ID.into(),
+                stream: false,
+                ..Default::default()
+            });
+        then.json(chat_completions_response.clone());
+    });
+
+    // Start orchestrator server and its dependencies
+    let mock_detector_server = MockServer::new(detector_name).with_mocks(detector_mocks);
+    let mock_chat_completions_server = MockServer::new("chat_completions").with_mocks(chat_mocks);
+
+    let orchestrator_server = TestOrchestratorServer::builder()
+        .config_path(ORCHESTRATOR_CONFIG_FILE_PATH)
+        .detector_servers([&mock_detector_server])
+        .chat_generation_server(&mock_chat_completions_server)
+        .build()
+        .await?;
+
+    // Make orchestrator call for input/output no detections
+    let response = orchestrator_server
+        .post(ORCHESTRATOR_CHAT_COMPLETIONS_DETECTION_ENDPOINT)
+        .json(&ChatCompletionsRequest {
+            model: MODEL_ID.into(),
+            detectors: Some(DetectorConfig {
+                input: Some(HashMap::from([(
+                    detector_name.into(),
+                    DetectorParams::new(),
+                )])),
+                output: Some(HashMap::from([(
+                    detector_name.into(),
+                    DetectorParams::new(),
+                )])),
+            }),
+            messages,
+            ..Default::default()
+        })
+        .send()
+        .await?;
+
+    // Assertions for no detections
+    assert_eq!(response.status(), StatusCode::OK);
+    let results = response.json::<ChatCompletion>().await?;
+    assert_eq!(results.choices[0], chat_completions_response.choices[0]);
+    assert_eq!(results.choices[1], chat_completions_response.choices[1]);
+    assert_eq!(results.warnings, vec![]);
+    assert_eq!(results.detections, expected_detections.clone());
```

Comment:
Unneeded clone as `expected_detections` will not be used after this line.
```suggestion
    assert_eq!(results.detections, expected_detections);
```